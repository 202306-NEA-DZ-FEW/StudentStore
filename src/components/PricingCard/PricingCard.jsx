import React from "react";
import { toast } from "react-toastify";

import Button from "../Buttons/Button";

const PricingCard = ({ title, amount }) => {
    const handleClick = () => {
        toast.success(`Thank you for donating ${amount}$`);
    };
    return (
        <div className='p-8  md:p-4 text-black text-center border rounded-lg shadow-md bg-white'>
            <h2 className='text-xl   font-semibold'>{title}</h2>
            <p className=' text-6xl my-2 font-semibold flex justify-center'>
                <span className='text-3xl self-center'>$</span>
                {amount}
            </p>
            <Button onClick={handleClick}>Donate</Button>
        </div>
    );
};

export default PricingCard;
