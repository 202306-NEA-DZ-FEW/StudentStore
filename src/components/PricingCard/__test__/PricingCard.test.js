import renderer from "react-test-renderer";

import PricingCard from "../PricingCard";

it("renders correctly", () => {
    const tree = renderer.create(<PricingCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
