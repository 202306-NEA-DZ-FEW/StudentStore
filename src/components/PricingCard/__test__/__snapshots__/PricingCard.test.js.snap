// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingCard renders correctly 1`] = `
<div
  className="p-8 md:p-4 text-black text-center border rounded-lg shadow-md bg-white"
>
  <h2
    className="text-xl   font-semibold"
  >
    Small help
  </h2>
  <p
    className=" text-6xl my-2 font-semibold flex justify-center"
  >
    <span
      className="text-3xl self-center"
    >
      $
    </span>
    10
  </p>
  <button
    className="bg-[#FF8A57] border-2 border-[#FF8A57] py-2 rounded-lg text-white shadow-sm text-md hover:bg-white hover:text-[#FF8A57] hover:border-2 hover:border-[#FF8A57] hover:duration-300 font-bold w-36 md:w-44 mx-auto px-0"
    onClick={[Function]}
  >
    donate
  </button>
</div>
`;
